ml_project/src/features/build_features.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+import sys
+from typing import List, Optional
+import click
+import logging
+from pathlib import Path
+import json
+
+import pandas as pd
+
+try:
+    from ml_project.entities import read_training_pipeline_params
+except ModuleNotFoundError:
+    project_dir = Path(__file__).resolve().parents[3]
+    sys.path.append(str(project_dir))
+    from ml_project.entities import read_training_pipeline_params
+
+# from dotenv import find_dotenv, load_dotenv
+
+INTERIM_DATA_FILEPATH = './data/interim'
+PROCESSED_DATA_FILEPATH = './data/processed'
+
+
+class CatFeatureTargetEncoder:
+    def __init__(
+            self,
+            cat_feature_list: List[str],
+            target_name: str,
+            logger: logging.Logger,
+            min_count: Optional[int] = None,
+            inplace: bool = False
+    ):
+        self.cat_feature_list = cat_feature_list
+        self.target_name = target_name
+        self.inplace = inplace
+        self.logger = logger
+        self.encoded_dicts = dict()
+        self.min_count = min_count
+
+    def fit(self, dataset):
+        for feature_name in self.cat_feature_list:
+            encoded_dict = self._build_encoded_dict(dataset, feature_name)
+            self.encoded_dicts.update(encoded_dict)
+
+        self.logger.info(f'{id(self)} fitted')
+
+        return self
+
+    def transform(self, dataset):
+        dataset_ = dataset.copy()
```

Comment:
add check fir or not
![image](https://user-images.githubusercontent.com/49230518/170886528-eba2f995-b4d9-40b7-9583-d06579f2b887.png)

Reply:
done